c7n/resources/rds.py

```diff
@@ -1454,8 +1454,9 @@ def get_permissions(self):
 
     def process(self, configs, event=None):
         rds = self.manager.get_resource_manager('rds').resources()
-        self.used = {r.get('DBSubnetGroupName', r['DBInstanceIdentifier'])
-                     for r in rds}
+        self.used = {
+            r.get('DBSubnetGroup', r['DBInstanceIdentifier']).get('DBSubnetGroupName')
```

Comment:
the r['DBInstanceIdentifier'] here is strange..  i think we should just do a jmespath expression
` set(jmespath.search('DBSubnetGroup.DBSubnetGroupName', rds))`

there's an open question here on aurora serverless as well, as i'm not sure it would show up without explicitly querying them, for normal aurora clusters there are db instances behind them.

Reply:
Good call on jmespath search. I am unsure of why DB identifier was used there originally, it does seem like a strange fall back.

Reply:
I just checked on aurora normal and aurora serverless and they apparently need to be retrieved via a describe cluster call. I added that here on the rds filter for full coverage. Since these are registered as two different resources, do you think I should also add from the cluster side?

Reply:
aurora normal will populate rds instances as cluster members, so their covered afaict, aurora serverless is distinct. i would just pull the resource manager for aurora / rds-cluster to query their values in this.

Reply:
Ah nice! Much cleaner!! Added in 1dfa143